### DISCUSSION TITLE
Supabase Auth - Nextjs Pages Router - Session coming back null upon signin

### DISCUSSION THREAD

**Discussion by slawton3** [supabase/supabase#16825] [Contribute to Supabase]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

When I sign into my application, the session fetched on the server via ```getServerSideProps``` is coming back as null and redirects to my signin page.

```
const Home: NextPage = () => {
  return (
    <Layout>
      <HomeView />
    </Layout>
  );
};

export default Home;

export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
  // Create authenticated Supabase Client
  const supabase = createPagesServerClient(ctx);
  // Check if we have a session
  const {
    data: { session },
  } = await supabase.auth.getSession();

  console.log(session);  //returns null even after successfully signing in

  if (!session)
    return {
      redirect: {
        destination: AppRoutes[RouteName.SignIn].Path,
        permanent: false,
      },
    };

  return {
    props: {
      initialSession: session,
      user: session.user,
    },
  };
};

```

I have verified that others are getting the same issue and I have not found a resolution for this yet.

## Expected behavior

When I sign in, the session should be available and I should be able to see my home page.

## Additional context

Nextjs version: 13.1

"@supabase/auth-helpers-nextjs": "^0.7.2",
"@supabase/auth-helpers-react": "^0.4.0",
"@supabase/auth-ui-react": "^0.4.2",
"@supabase/auth-ui-shared": "^0.1.6",
"@supabase/supabase-js": "^2.24.0"

**Comment by KSSaiTeja** [comment]
can i work on this issue

**Comment by saltcod** [comment]
Sure @KSSaiTeja !

**Comment by saltcod** [comment]
@slawton3 Have you followed a guide for this? If you've got that handy, might be useful for helping you debug.
